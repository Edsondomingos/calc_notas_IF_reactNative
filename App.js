import * as React from 'react';
import { NavigationContainer } from '@react-navigation/native'
import { createNativeStackNavigator } from '@react-navigation/native-stack'
import Home from './components/Home'
import CalcMedioTecnico from  './components/CalcMedioTecnico'
import CalcSuperior from  './components/CalcSuperior'
import Sobre from './components/Sobre'

const Stack = createNativeStackNavigator()

export default function App() {
  return (
    <NavigationContainer>
      <Stack.Navigator  initialRouteName='Home' >
        <Stack.Screen name='Home' component={Home} />
        <Stack.Screen name='CalcMedioTecnico' component={CalcMedioTecnico} />
        <Stack.Screen name='CalcSuperior' component={CalcSuperior} />
        <Stack.Screen name='Sobre' component={Sobre} />
      </Stack.Navigator>
    </NavigationContainer>
  );
}
