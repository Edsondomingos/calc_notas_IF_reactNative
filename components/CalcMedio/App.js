import * as React from 'react';
import { Text, View, StyleSheet, Button, TextInput, TouchableHighlight } from 'react-native';
import {useState }from "react";
//import Estilo from './src/style';
let notafinal

export default () => {
const [nota1, nota1SetNumero] = useState('');
const [nota2, nota2SetNumero] = useState('');
const [nota3, nota3SetNumero] = useState('');
const [nota4, nota4SetNumero] = useState('');
const [notafinal2, finalsetNumero] = useState('');
const [mediaFinal, mediaFinalSetNumero] = useState('');

function result (){
  let n1 = parseFloat(nota1)
  let n2 = parseFloat(nota2)
  let n3 = parseFloat(nota3)
  let n4 = parseFloat(nota4)
  if( nota1.length == 0 ){
      return
    }
//mediaFinalSetNumero (parseFloat(nota1 * 2 + nota2 * 2 + nota3 * 3 + nota4 * 3) / 10);
  if( nota2.length == 0 ){
    notafinal = (((300 - (n1*2))/2)) 
    finalsetNumero(notafinal)
  }  
    if (nota1.length != 0 && nota2.length != 0 && nota3.length != 0 && nota4.length != 0){
      notafinal = (((n1 * 2) + (n2 * 2) + (n3 * 3) + (n4 * 3)) / 10)
      if(notafinal < 50){
        return alert('Burro')
      } 
      mediaFinalSetNumero(notafinal), alert ('Arrasou')
      
    }
  }    


return (
  <View>
    <View>
      <Text>Calculadora Medio</Text>
    </View>
    <View>
      <Text>Digite a nota:</Text>
      <TextInput value={nota1} placeholder='Nota 1' onChangeText={nota1SetNumero}/>
      <Text>Digite a nota:</Text>
      <TextInput value={nota2} placeholder='Nota 2' onChangeText={nota2SetNumero}/>
      <Text>Digite a nota:</Text>
      <TextInput value={nota3} placeholder='Nota 3' onChangeText={nota3SetNumero}/>
      <Text>Digite a nota:</Text>
      <TextInput value={nota4} placeholder='Nota 4' onChangeText={nota4SetNumero}/>
      
      <TextInput placeholder='falta' value={notafinal2} onChangeText={finalsetNumero} />
        
        <Button title="Calcular" onPress={() => {result()}} />

    </View>
    <Text>Resultado: {mediaFinal}</Text>
  </View>
  );
}

