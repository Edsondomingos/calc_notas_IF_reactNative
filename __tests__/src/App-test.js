import renderer from 'react-test-renderer'
import App from '../../App';

describe('Teste de snapshot', () => {
    it('Snapshot de App', () => {
        const tela = renderer.create(<App />).toJSON()
        expect(tela).toMatchSnapshot()
    })    
})