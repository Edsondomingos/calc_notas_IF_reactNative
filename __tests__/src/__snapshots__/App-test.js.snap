// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Teste de snapshot Snapshot de App 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      stackPresentation="push"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="Home"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "backgroundColor": "rgb(242, 242, 242)",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "flex": 1,
            }
          }
        >
          <Image
            source={1}
            style={
              Object {
                "marginHorizontal": "auto",
              }
            }
          />
          <Text
            style={
              Object {
                "fontFamily": "Cursive",
                "fontSize": 30,
                "fontWeight": "bold",
                "marginVertical": 20,
                "textAlign": "center",
              }
            }
          >
            Calculadora de notas
          </Text>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#f05",
                "borderBottomEndRadius": 20,
                "borderTopEndRadius": 20,
                "borderWidth": 1,
                "height": 70,
                "justifyContent": "center",
                "margin": "auto",
                "marginVertical": 30,
                "padding": 10,
                "width": 200,
              }
            }
            testID="btnTelaCalcMedio"
          >
            <Text
              style={
                Object {
                  "color": "#fff",
                  "fontFamily": "Cursive",
                  "fontSize": 20,
                  "fontWeight": "bold",
                }
              }
            >
              Médio➗Técnico
            </Text>
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#f05",
                "borderBottomEndRadius": 20,
                "borderTopEndRadius": 20,
                "borderWidth": 1,
                "height": 70,
                "justifyContent": "center",
                "margin": "auto",
                "marginVertical": 30,
                "padding": 10,
                "width": 200,
              }
            }
            testID="btnTelaCalcSuperior"
          >
            <Text
              style={
                Object {
                  "color": "#fff",
                  "fontFamily": "Cursive",
                  "fontSize": 20,
                  "fontWeight": "bold",
                }
              }
            >
              ➕Superior
            </Text>
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#f05",
                "borderBottomEndRadius": 20,
                "borderTopEndRadius": 20,
                "borderWidth": 1,
                "height": 70,
                "justifyContent": "center",
                "margin": "auto",
                "marginVertical": 30,
                "padding": 10,
                "width": 200,
              }
            }
            testID="btnTelaSobre"
          >
            <Text
              style={
                Object {
                  "color": "#fff",
                  "fontFamily": "Cursive",
                  "fontSize": 20,
                  "fontWeight": "bold",
                }
              }
            >
              ❕Sobre
            </Text>
          </View>
        </View>
      </View>
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
